### PR TITLE
Add seperate properties for nginx process user/group

### LINKED ISSUE #572: Incorrect permissions of nginx config files
## :ghost: Brief Description

The `nginx_config` resource, which is responsible of the creation of `nginx.conf` and other configuration folders sets the uid and gid based on `nginx_user` and `nginx_group` functions, which returns users such as `www-data` or `nobody` depending on the platform. I believe that the owner of the configuration files should always be `root`.

## :pancakes: Cookbook version

v11.2.0

## :woman_cook:  Chef-Infra Version

16.9.20

## :tophat: Platform details

Ubuntu 20.04, but should be able to be replicated elsewhere.

## Steps To Reproduce

Converge a recipe as the [documentation says](https://github.com/sous-chefs/nginx/blob/master/documentation/nginx_config.md#examples) in an Ubuntu 20.04 Dokken image:

```ruby
nginx_config 'nginx' do
  action :create
  notifies :reload, 'nginx_service[nginx]', :delayed
end
```

You should see the following:

```bash
root@dokken:/etc/nginx# ll
total 56
drwxr-xr-x 4 root     root     4096 Feb 10 16:23 ./
drwxr-xr-x 1 root     root     4096 Feb 10 16:19 ../
drwxr-x--- 2 www-data www-data 4096 Feb 10 16:19 conf.d/
drwxr-x--- 2 www-data www-data 4096 Feb 10 16:23 conf.http.d/
-rw-r--r-- 1 root     root     1007 Apr 21  2020 fastcgi_params
-rw-r--r-- 1 root     root     2837 Apr 21  2020 koi-utf
-rw-r--r-- 1 root     root     2223 Apr 21  2020 koi-win
-rw-r--r-- 1 root     root     5231 Apr 21  2020 mime.types
lrwxrwxrwx 1 root     root       22 Oct 29 09:59 modules -> /usr/lib/nginx/modules/
-rw-r----- 1 www-data www-data  806 Feb 10 16:23 nginx.conf
-rw-r--r-- 1 root     root      636 Apr 21  2020 scgi_params
-rw-r--r-- 1 root     root      664 Apr 21  2020 uwsgi_params
-rw-r--r-- 1 root     root     3610 Apr 21  2020 win-utf
```

## :police_car: Expected behavior

The `nginx.conf` configuration file and `conf.d` / `conf.http.d` folders should be owned by `root`.

## :heavy_plus_sign: Additional context

N/A

### PR DESCRIPTION
**Pull request by bmhughes**
# Description

Add seperate properties for nginx process user/group so nginx configuration file/folder owner/group does not effect the run-as user/group.

## Issues Resolved

- Fixes #572 

## Check List

- [x] All tests pass. See TESTING.md for details.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by ramereth** [comment]
@bmhughes I almost feel like this should be a major bump since you're removing/changing resource properties but I'm not sure how others feel

**Comment by bmhughes** [comment]
> @bmhughes I almost feel like this should be a major bump since you're removing/changing resource properties but I'm not sure how others feel

Yes you're 100% right, that's passed me by somehow. I think a property alias should keep it clean without a major version jump and can be removed when it does jump a major version again.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ This file is used to list changes made in each version of the nginx cookbook.
 
 ## Unreleased
 
+- Add seperate properties for nginx process user/group (Issue #572)
```
Comment (ramereth) [inline CHANGELOG.md:7]: Please add a note here that you're deprecating the other property and also mention which resource you're changing here.

**Review by ramereth — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (bmhughes) [reply CHANGELOG.md]: Done

**Review by ramereth — APPROVED** [review]
@Xorima could you approve too please if you're OK?

**Review by Xorima — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [11.3.0](https://github.com/sous-chefs/nginx/releases/tag/11.3.0)

**Pull request merged**